oasislmf/pytools/common/data.py

```diff
@@ -141,3 +156,183 @@ def load_as_array(dir_path, name, _dtype, must_exist=True):
+        event_rates (ndarray[oasis_float]): event rates
+    """
+    try:
+        event_rate_file = Path(run_dir, filename)
+        data = load_as_ndarray(run_dir, filename[:-4], np.dtype([('event_id', oasis_int), ('rate', oasis_float)]),
+                               col_map={"event_id": "EventIds", "rate": "Event_rates"})
+        if data is None or data.size == 0:
+            logger.info(f"Event rate file {event_rate_file} is empty, proceeding without event rates.")
+            return np.array([], dtype=oasis_int), np.array([], dtype=oasis_float)
+        unique_event_ids = data['event_id']
+        event_rates = data['rate']
+
+        # Make sure event_ids are sorted
+        sort_idx = np.argsort(unique_event_ids)
+        unique_event_ids = unique_event_ids[sort_idx]
+        event_rates = event_rates[sort_idx]
+        return unique_event_ids, event_rates
+    except Exception as e:
+        logger.warning(f"An error occurred while reading event rate file: {str(e)}")
+        return np.array([], dtype=oasis_int), np.array([], dtype=oasis_float)
+
+
+def read_quantile(sample_size, run_dir, filename=QUANTILE_FILE, return_empty=False):
+    """Generate a quantile interval Dictionary based on sample size and quantile binary file
+    Args:
+        sample_size (int): Sample size
+        run_dir (str | os.PathLike): Path to input files dir
+        filename (str | os.PathLike): quantile binary file name
+        return_empty (bool): return an empty intervals array regardless of the existence of the quantile binary
+    Returns:
+        intervals (quantile_interval_dtype): Numpy array emulating a dictionary for numba
+    """
+    intervals = []
+    quantile_interval_dtype = np.dtype([
+        ('q', oasis_float),
+        ('integer_part', oasis_int),
+        ('fractional_part', oasis_float),
+    ])
+
+    if return_empty:
+        return np.array([], dtype=quantile_interval_dtype)
+    try:
+        data = load_as_ndarray(run_dir, filename[:-4], np.float32)
+        for q in data:
+            # Calculate interval index and fractional part
+            pos = (sample_size - 1) * q + 1
+            integer_part = int(pos)
+            fractional_part = pos - integer_part
+            intervals.append((q, integer_part, fractional_part))
+    except Exception as e:
+        raise RuntimeError(f"An error occurred: {str(e)}")
+
+    # Convert to numpy array
+    intervals = np.array(intervals, dtype=quantile_interval_dtype)
+    return intervals
+
+
+def read_occurrence(run_dir, filename=OCCURRENCE_FILE):
+    """Read the occurrence binary file and returns an occurrence map
+    Args:
+        run_dir (str | os.PathLike): Path to input files dir
+        filename (str | os.PathLike): occurrence binary file name
+    Returns:
+        occ_map (ndarray[occ_map_dtype]): numpy map of event_id, period_no, occ_date_id from the occurrence file
+    """
+    from .event_stream import mv_read
```

Comment:
Because you have to import here to avoid circular import, I would put all the specific file reader in an other python file (this is a personal preference here, so as you like, totally ok if you prefer it like this).